nextcloudappstore/core/static/assets/js/app-upload.js

```diff
@@ -0,0 +1,132 @@
+(function (window) {
+    'use strict';
+
+
+    function apiRequestToken(csrf) {
+        var request = new Request(
+            '/api/v1/token',
+            {
+                method: 'POST',
+                headers: new Headers({
+                    'Content-Type': 'application/json',
+                    'X-CSRFToken': csrf,
+                }),
+                credentials: 'include'
+            }
+        );
+
+        return fetch(request).then((response) => {
```

Comment:
What are you trying to do here? Why not simply 
    
    return fetch(request).then((response) => response.json());

if its not 200 its rejected by default afaik